src/arch/ArchHooks/ArchHooks_Win32Static.cpp

```diff
@@ -9,38 +9,45 @@
 #include <cstdint>
 #include <vector>
 
-// for timeGetTime
+// for QueryPerformanceCounter
 #include <windows.h>
 #include <mmsystem.h>
 #if defined(_MSC_VER)
 #pragma comment(lib, "winmm.lib")
 #endif
 
-
 static bool g_bTimerInitialized;
 
+/* QueryPerformanceCounter variables below.
+ * QueryPerformanceCounter and QueryPerformanceFrequency expect
+ * a LARGE_INTEGER, which is a union. These functions store data
+ * in the QuadPart of the LARGE_INTEGER, which is a 64-bit integer. */
+LARGE_INTEGER g_bFrequency, g_bStartTime, g_bCurrentTime;
+
 static void InitTimer()
 {
 	if( g_bTimerInitialized )
 		return;
 	g_bTimerInitialized = true;
 
-	timeBeginPeriod( 1 );
+	// Grab important information for QPC during the timer initialization
+	QueryPerformanceCounter(&g_bStartTime);
+	QueryPerformanceFrequency(&g_bFrequency);
 }
 
-std::int64_t ArchHooks::GetMicrosecondsSinceStart( bool bAccurate )
+std::int64_t ArchHooks::GetMicrosecondsSinceStart(bool bAccurate)
 {
-	if( !g_bTimerInitialized )
+	// Make sure the timer is initialized
+	if (!g_bTimerInitialized)
 		InitTimer();
 
-	std::int64_t ret = timeGetTime() * std::int64_t(1000);
-	if( bAccurate )
-	{
-		ret = FixupTimeIfLooped( ret );
-		ret = FixupTimeIfBackwards( ret );
-	}
+	// Get the current time
+	QueryPerformanceCounter(&g_bCurrentTime);
 
-	return ret;
+	/* Calculate the elapsed time in microseconds.
+	 * Add 0.49 to round and cast at the same time, to ensure the value is rounded
+	 * to the nearest integer. Maintaning accuracy here is extremely important. */
+	return static_cast<double>(g_bCurrentTime.QuadPart - g_bStartTime.QuadPart) * 1000000.0 / static_cast<double>(g_bFrequency.QuadPart) + 0.49;
```

Comment:
My previous comment said we don't need to cast to double anymore as per the actual microsoft documentation. Refer to that comment and make the changes accordingly.

Reply:
> My previous comment said we don't need to cast to double anymore as per the actual microsoft documentation. Refer to that comment and make the changes accordingly.

Accuracy is reduced to 0.3ms std dev*3 if we change this line to `return (g_bCurrentTime.QuadPart - g_bStartTime.QuadPart) * 1000000.0 / g_bFrequency.QuadPart;`, but I will commit that if you'd prefer it.

![625BPM_SYN_2024-05-04_021301](https://github.com/itgmania/itgmania/assets/163092272/ec0c0b8a-c8a1-48f1-8a37-d854ca628ea3)